### PR TITLE
fix(styles): update shadowdom prefix styles with hx

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: SURF-1126

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

File: src/helix-ui/elements/HXAlertElement.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div id="wrapper">
+<div id="hxWrapper">
   <hx-icon id="icon" type="info-circle"></hx-icon>
   <span id="content">
```
Comment (CITguy) [inline src/helix-ui/elements/HXAlertElement.html]: Every `id` and `class` used in the shadow dom will require prefixing, not just the root element.

File: src/helix-ui/elements/HXCheckboxElement.less [hunk 2]
```diff
@@ -33,23 +33,23 @@
 }
 
 /* icons */
-#minus,
-#tick {
+#hxMinus,
+#hxTick {
```
Comment (CITguy) [inline src/helix-ui/elements/HXCheckboxElement.less]: nest these within `#hxCustomControl`

File: src/helix-ui/elements/HXFileIconElement.less [hunk 3]
```diff
@@ -1,31 +1,31 @@
 @import './HXElement';
 @import 'elements/hx-icon';
 
-#wrapper {
+#hxFileIcon{
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileIconElement.less]: add space before brace

File: src/helix-ui/elements/HXModalElement.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ export class HXModalElement extends HXElement {
 
     connectedCallback () {
         this.$upgradeProperty('open');
-        this._btnClose = this.shadowRoot.querySelector("#close");
+        this._btnClose = this.shadowRoot.querySelector("#hxClose");
```
Comment (CITguy) [inline src/helix-ui/elements/HXModalElement.js]: Please change this to a `get _btnClose () { ... }`

File: src/helix-ui/elements/HXSearchElement.html [hunk 5]
```diff
@@ -1,10 +1,10 @@
-<label id="wrapper">
-  <input type="text" role="search" id="search" autocomplete="off" />
-  <button type="button" id="clear" hidden aria-label="Clear search">
+<label id="hxSearch">
+  <input type="text" role="search" id="hxSearchInput" autocomplete="off" />
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.html]: `hxSearchInput` --> `hxNativeControl`

File: src/helix-ui/elements/HXSearchElement.js [hunk 6]
```diff
@@ -60,8 +60,8 @@ export class HXSearchElement extends HXElement {
 
     constructor () {
         super();
-        this._elSearch = this.shadowRoot.getElementById('search');
-        this._btnClear = this.shadowRoot.getElementById('clear');
+        this._elSearch = this.shadowRoot.getElementById('hxSearchInput');
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.js]: convert to `get _elSearch () { ... }`

File: src/helix-ui/elements/HXSearchElement.js [hunk 7]
```diff
@@ -60,8 +60,8 @@ export class HXSearchElement extends HXElement {
...
+        this._btnClear = this.shadowRoot.getElementById('hxClear');
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.js]: convert to `get _btnClear () { ... }`

File: src/helix-ui/elements/HXSearchElement.less [hunk 8]
```diff
@@ -23,7 +23,7 @@
     z-index: 1;
   }
 
-  #search {
+  #hxSearchInput {
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.less]: `#hxSearchInput` --> `#hxNativeControl`

File: src/helix-ui/elements/HXSearchElement.less [hunk 9]
```diff
@@ -85,48 +85,48 @@
...
       }
 
-      ~ #customControl {
+      ~ #hxCustomControl {
         border-color: @cyan-700;
         box-shadow: @focus-glow;
       }
     }//:focus
-  }//#clear
+  }//#hxClear
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.less]: remove comment

File: src/helix-ui/elements/HXSearchElement.less [hunk 10]
```diff
@@ -85,48 +85,48 @@
...
-  #clear:focus {
+  #hxClear:focus {
     hx-icon {
       outline-color: currentColor;
     }
   }
 
-  #search:focus {
-    ~ #clear {
+  #hxSearchInput:focus {
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.less]: `#hxSearchInput` --> `#hxNativeControl`

File: src/helix-ui/elements/HXSearchElement.less [hunk 11]
```diff
@@ -85,48 +85,48 @@
...
   }
 
   // make sure that our custom control doesn't lose focus style
-  // when we tab between #search and #clear
-  #clear:focus,
-  #search:focus {
-    ~ #customControl {
+  // when we tab between #hxSearch and #hxClear
+  #hxClear:focus,
+  #hxSearchInput:focus {
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.less]: `#hxSearchInput` --> `#hxNativeControl`

File: src/helix-ui/elements/HXSearchElement.less [hunk 12]
```diff
@@ -136,26 +136,26 @@
...
     color: inherit;
   }
 
-  #clear {
+  #hxClear {
     display: none;
   }
 
-  #search {
+  #hxSearchInput {
```
Comment (CITguy) [inline src/helix-ui/elements/HXSearchElement.less]: `#hxSearchInput` --> `#hxNativeControl`

File: src/helix-ui/elements/HXToastElement.html [hunk 13]
```diff
@@ -1,14 +1,14 @@
-<div id="wrapper">
-  <div id="icon-wrapper">
-    <hx-icon id="icon" type="info-circle"></hx-icon>
+<div id="hxToast">
+  <div id="hxIcon-wrapper">
```
Comment (CITguy) [inline src/helix-ui/elements/HXToastElement.html]: `#hxIcon-wrapper` --> `#hxIconWrapper`

File: src/helix-ui/elements/HXToastElement.less [hunk 14]
```diff
@@ -32,89 +32,89 @@
...
-}
-
-#icon-wrapper {
-  align-items: center;
-  display: flex;
-  margin: 0 0.75rem 0 0.5rem;
 
-  hx-icon {
-    font-size: 2rem;
+  #hxIcon-wrapper {
```
Comment (CITguy) [inline src/helix-ui/elements/HXToastElement.less]: `#hxIcon-wrapper` --> `#hxIconWrapper`

File: src/helix-ui/elements/HXToastElement.less [hunk 15]
```diff
@@ -32,89 +32,89 @@
...
 :host([type="info"]) {
-  #wrapper {
+  #hxToast {
     &::after {
       border-left-color: @purple-500;
     }
   }
 
-  #icon {
+  #Icon {
```
Comment (CITguy) [inline src/helix-ui/elements/HXToastElement.less]: `#Icon` --> `#hxIcon`

File: src/helix-ui/elements/HXToastElement.less [hunk 16]
```diff
@@ -32,89 +32,89 @@
...
 :host([type="error"]) {
-  #wrapper {
+  #hxToast {
     &::after {
       border-left-color: @red-900;
     }
   }
 
-  #icon {
+  #Icon {
```
Comment (CITguy) [inline src/helix-ui/elements/HXToastElement.less]: `#Icon` --> `#hxIcon`

File: src/helix-ui/elements/HXToastElement.less [hunk 17]
```diff
@@ -32,89 +32,89 @@
...
 :host([type="success"]) {
-  #wrapper {
+  #hxToast {
     &::after {
       border-left-color: @green-500;
     }
   }
 
-  #icon {
+  #Icon {
```
Comment (CITguy) [inline src/helix-ui/elements/HXToastElement.less]: `#Icon` --> `#hxIcon`

File: src/helix-ui/elements/HXSearchElement.less [hunk 18]
```diff
@@ -12,13 +12,13 @@ input::-ms-clear {
   height: 2rem;
```
Comment (nimbinatus) [inline src/helix-ui/elements/HXSearchElement.less]: For line 6: Update to ensure input never gets a margin applied. Ref #212

Comment (nimbinatus) [reply src/helix-ui/elements/HXSearchElement.less]: Decided to make a fast follow after merge.

**Pull request merged**